internal/client/concurrency_limits.go

```diff
@@ -0,0 +1,95 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+)
+
+var _ = api.ConcurrencyLimitsClient(&ConcurrencyLimitsClient{})
+
+// ConcurrencyLimitsClient is a client for working with concurrency limits.
+type ConcurrencyLimitsClient struct {
+	hc          *http.Client
+	routePrefix string
+	apiKey      string
+}
+
+// ConcurrencyLimits returns a ConcurrencyLimitsClient.
+//
+//nolint:ireturn // required to support PrefectClient mocking
+func (c *Client) ConcurrencyLimits(accountID uuid.UUID, workspaceID uuid.UUID) (api.ConcurrencyLimitsClient, error) {
+	if accountID == uuid.Nil {
+		accountID = c.defaultAccountID
+	}
+
+	if workspaceID == uuid.Nil {
+		workspaceID = c.defaultWorkspaceID
+	}
+
+	if err := validateCloudEndpoint(c.endpoint, accountID, workspaceID); err != nil {
+		return nil, err
+	}
+
+	return &ConcurrencyLimitsClient{
+		hc:          c.hc,
+		routePrefix: getWorkspaceScopedURL(c.endpoint, accountID, workspaceID, "concurrency_limits"),
+		apiKey:      c.apiKey,
+	}, nil
+}
+
+// Create creates a new concurrency limit.
+func (c *ConcurrencyLimitsClient) Create(ctx context.Context, data api.ConcurrencyLimitCreate) (*api.ConcurrencyLimit, error) {
+	cfg := requestConfig{
+		method:       http.MethodPost,
+		url:          c.routePrefix,
```

Comment:
```suggestion
		url:          c.routePrefix + "/",
```

Small change here related to https://linear.app/prefect/issue/PLA-948/workspace-create-fails-with-404-when-endpoint-is-set